dao/authentication_dao_test.go

```diff
@@ -0,0 +1,312 @@
+package dao
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/RedHatInsights/sources-api-go/marketplace"
+	"github.com/RedHatInsights/sources-api-go/redis"
+	"github.com/hashicorp/vault/api"
+	"github.com/labstack/gommon/log"
+	"github.com/sirupsen/logrus"
+)
+
+func TestMain(t *testing.M) {
+	// we need this to parse arguments otherwise there are not recognized which lead to error
+	flag.Bool("createdb", false, "create the test database")
+	flag.Bool("integration", false, "run unit or integration tests")
+	flag.Parse()
+
+	os.Exit(t.Run())
```

Comment:
I _think_ we can use the parser.ParseFlags() function for this right?

Reply:
You are right. I think I forgot to update this one on the rebase :roll_eyes: